taqueria-vscode-extension-web-ui/src/App.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { AllEditors } from './AllEditors';
+import { MichelineEditor } from './data-editors/MichelineEditor';
+import { InteropMessageInterface } from './interopTypes';
+
+const state = {
+	interop: undefined as undefined | InteropMessageInterface,
+	onChangeCallback: (x: InteropMessageInterface) => {/* empty */},
+	logs: [] as { message: string; data?: unknown }[],
+};
+
+declare global {
+	interface Window {
+		setGlobalInteropMessageInterface: (x: InteropMessageInterface) => void;
+	}
+}
+
+// const log = (message: string, data?: unknown) => {
+// 	state.logs.push({ message, data });
+// };
+// console.log = log;
+
+const setGlobalInteropMessageInterface = (x: InteropMessageInterface) => {
+	state.interop = x;
+	state.onChangeCallback(x);
+};
+window.setGlobalInteropMessageInterface = setGlobalInteropMessageInterface;
+
+export const App = () => {
+	const [deps, setDeps] = useState(state.interop);
+	useEffect(() => {
+		state.onChangeCallback = x => {
+			setDeps(x);
+		};
+		return () => {
+			state.onChangeCallback = () => {};
+		};
+	}, []);
+
+	console.log('App RENDER', { deps, interop: state.interop });
+
+	return (
+		<>
+			<EditorSelector deps={deps} />
+			<DebugView />
+		</>
+	);
+};
+
+const EditorSelector = ({ deps }: { deps: undefined | InteropMessageInterface }) => {
+	if (deps?.view === 'AllEditors') {
+		return <AllEditors />;
+	}
+	if (deps?.view === 'MichelineEditor') {
+		return <MichelineEditor {...deps} />;
+	}
+
+	return <DefaultView />;
+};
+
+const DebugView = () => {
+	const DEBUG = true;
+	if (!DEBUG) {
+		return <></>;
```

Comment:
Why empty fragment? Should be `null` I think

Reply:
@ricklove